src/features/domain-settings/hooks/useDomainSettingsForm.tsx

```diff
@@ -0,0 +1,227 @@
+	stepId: string;
+	formHeader: string;
+	formContent: ReactNode;
+	isTransactionLoading: boolean;
+	onStepBarUpdate: (step: Step) => void;
+};
+
+export const useDomainSettingsForm = (
+	zna: string,
+	onClose: () => void,
+): UseDomainSettingsFormFormReturn => {
+	const sdk = useZnsSdk();
+
+	const { account, provider } = useWeb3();
+	const { executeTransaction } = useTransaction();
+	const { domainId, metadata, isMetadataLocked } = useDomainSettingsData(zna);
+
+	const [stepId, setStepId] = useState(steps[0].id);
+	const [errorText, setErrorText] = useState<string>();
+	const [details, setDetails] = useState<DetailsFormSubmit>();
+	const [loadingStatusText, setLoadingStatusText] = useState<string>();
+	const [formHeader, setFormHeader] = useState<string>('My Domain Settings');
+	const [isTransactionLoading, setIsTransactionLoading] = useState<boolean>();
+	const [confirmActionType, setConfirmActionType] =
+		useState<ConfirmActionType>();
+
+	const onStepBarUpdate = (step: Step) => {
+		setStepId(step.id);
+	};
+
+	const onRestart = () => {
+		setStepId(steps[0].id);
+	};
+
+	// Set metadata form details
+	const onSubmitMetadata = ({
+		title,
+		description,
+		isMintable,
+		isBiddable,
+		gridViewByDefault,
+		customDomainHeader,
+		customDomainHeaderValue,
+	}: FieldValues): void => {
+		setDetails({
+			attributes: metadata?.attributes,
+			title,
+			description,
+			image: metadata?.image,
+			image_full: metadata?.image_full,
+			previewImage: metadata?.previewImage,
+			animation_url: metadata?.animation_url,
+			stakingRequests: metadata?.stakingRequests,
+			isBiddable,
+			isMintable,
+			gridViewByDefault,
+			customDomainHeader,
+			customDomainHeaderValue,
+		});
+	};
+
+	const onCheckMetadataLockStatus = useCallback(async () => {
+		const isDomainMetadataLocked = await sdk.isDomainMetadataLocked(
+			domainId,
+			provider.getSigner(),
+		);
+
+		return isDomainMetadataLocked;
+	}, [domainId, provider]);
+
+	// Transaction handlers
+	const handleTransactionStart = (onLoadingHeader: string) => {
+		setIsTransactionLoading(true);
+		setLoadingStatusText('Waiting approval from your wallet...');
+		setFormHeader(onLoadingHeader);
+	};
+
+	const handleTransactionProcessing = (onLoadingText: string) => {
+		setLoadingStatusText(onLoadingText);
+	};
+
+	const handleTransactionSuccess = () => {
+		setIsTransactionLoading(false);
+		setStepId(steps[2].id);
+		onCheckMetadataLockStatus();
+	};
+
+	const handleTransactionError = (errorMessage: string) => {
+		setIsTransactionLoading(false);
+		setErrorText(errorMessage);
+		setStepId(stepId === FormStep.COMPLETE ? steps[2].id : steps[0].id);
+	};
+
+	const getSdkAction = (action: ConfirmActionType) => {
+		if (
+			action === ConfirmActionType.UNLOCK ||
+			action === ConfirmActionType.LOCK
+		) {
+			return sdk.lockDomainMetadata;
+		} else if (action === ConfirmActionType.SAVE_AND_LOCK) {
+			return sdk.setAndLockDomainMetadata;
+		} else {
+			return sdk.setDomainMetadata;
+		}
+	};
```

Comment:
nit: move this one outside of the hook since it doesn't rely on any state/props.